nilearn/experimental/surface/tests/test_surface_image.py

```diff
@@ -112,9 +112,93 @@ def test_load_save_mesh(
         assert np.array_equal(mesh.coordinates, expected_mesh.coordinates)
 
 
+def test_save_mesh_default_suffix(tmp_path, mini_img):
+    """Check default .gii extension is added."""
+    mini_img.mesh.to_filename(
+        tmp_path / "give_me_a_default_suffix_hemi-L_mesh"
+    )
+    assert (tmp_path / "give_me_a_default_suffix_hemi-L_mesh.gii").exists()
+
+
 def test_save_mesh_error(tmp_path, mini_img):
     with pytest.raises(ValueError, match="cannot contain both"):
-        mini_img.to_filename(tmp_path / "hemi-L_hemi-R_cannot_have_both.gii")
+        mini_img.mesh.to_filename(
+            tmp_path / "hemi-L_hemi-R_cannot_have_both.gii"
+        )
+
+
+def test_save_mesh_error_wrong_suffix(tmp_path, mini_img):
+    with pytest.raises(ValueError, match="with the extension '.gii'"):
+        mini_img.mesh.to_filename(
+            tmp_path / "hemi-L_hemi-R_cannot_have_both.foo"
+        )
+
+
+@pytest.mark.parametrize("use_path", [True, False])
+@pytest.mark.parametrize(
+    "output_filename, expected_files, unexpected_files",
+    [
+        ("foo.gii", ["foo_hemi-L.gii", "foo_hemi-L.gii"], ["foo.gii"]),
+        ("foo_hemi-L_T1w.gii", ["foo_hemi-L_T1w.gii"], ["foo_hemi-R_T1w.gii"]),
+        ("foo_hemi-R_T1w.gii", ["foo_hemi-R_T1w.gii"], ["foo_hemi-L_T1w.gii"]),
+    ],
+)
+def test_load_save_data(
+    tmp_path, output_filename, expected_files, unexpected_files, use_path
+):
+    mesh_right = datasets.fetch_surf_fsaverage().pial_right
```

Comment:
when cleaning the test suite we should have a fixture that creates very small gifti files to disk to test this with rather than relying on datasets.